darwin/Platform.c

```diff
@@ -168,6 +172,18 @@ bool Platform_init(void) {
    return true;
 }
 
+// Converts ticks in the Mach "timebase" to nanoseconds.
+// See `mach_timebase_info`, as used to define the `Platform_nanosecondsPerMachTick` constant.
+uint64_t Platform_machTicksToNanoseconds(uint64_t mach_ticks) {
+   uint64_t ticks_quot = mach_ticks / Platform_nanosecondsPerMachTickDenom;
+   uint64_t ticks_rem  = mach_ticks % Platform_nanosecondsPerMachTickDenom;
+
```

Comment:
```c
   assert(ticks_quot <= UINT64_MAX / Platform_nanosecondsPerMachTickNumer);
   assert(Platform_nanosecondsPerMachTickDenom <= UINT64_MAX / Platform_nanosecondsPerMachTickNumer);
```

These two constraints are what prevent the overflow from happening in this function. Not sure if these two assertions are worth adding into the function, but I mention these just for completeness sake.

The first assertion is related to a [time overflow](https://en.wikipedia.org/wiki/Time_formatting_and_storage_bugs) (like Y2K) that might be out of scope of htop.
The second assertion can always hold true when `mach_timebase_info_data_t` uses 32-bit integers for both numerator and denominator fields.

Reply:
They don't exactly hurt, but given this is what ultimately gives the time values used in htop, they should not straight up cause htop to terminate seemingly at random. Instead normal handling that causes a runtime warning (once) could be done instead …

Can you elaborate on the intuition of the second assertion?

Reply:
@BenBE 
```text
ticks_rem < Platform_nanosecondsPerMachTickDenom ;
ticks_rem * Platform_nanosecondsPerMachTickNumer < Platform_nanosecondsPerMachTickDenom * Platform_nanosecondsPerMachTickNumer

When `Platform_nanosecondsPerMachTickDenom * Platform_nanosecondsPerMachTickNumer` is less than 2^64, `ticks_rem * Platform_nanosecondsPerMachTickNumer` will be less than 2^64 as well, i.e. never overflows.

Reply:
Do you want me to do any changes?

The overflow should only happen when the result in nano-seconds overflows, which would already be a problem in other places in the application. This is after 584 CPU-years.

Reply:
Optional. I mostly asked for some clarifications here.

If you decided to update the PR, be sure to avoid direct `assert`s (for the reasons mentioned above), and instead choose a way that is as obvious as possible, that this is some intentional error value.

Reply:
@erikolofsson My view is that the time overflow bug that would happen 584 years later (the overflow on `part1`) is out of scope, but without an in-code comment, the `(ticks_rem * Platform_nanosecondsPerMachTickNumer)` in `part2` can be mistaken as prone to overflow. (It actually doesn't overflow for the reason I stated above.)

Reply:
I added your comment